tools/sigh.ts

```diff
@@ -318,17 +323,37 @@ function railroad(): boolean {
   return true;
 }
 
-function buildPath(path: string, deps?: string[]): () => boolean {
+function cleanObsolete() {
+  const exts = ['js', 'js.map', 'd.ts'];
+  for (const file of [...findProjectFiles('build', /javaharness|sigh\.js/, /\.js$/)]) {
```

Comment:
I don't think you need to turn this into an array, you can just iterate over it directly

Reply:
I do, actually - the loop removes files from the disk which causes problems in the readdir call.